### ISSUE TITLE
Sign in to Multiple Device Detection

### ISSUE THREAD

**Issue by xnicoting** [aws-amplify/amplify-android#2597]
**Amplify Categories:**
Authentication
DataStore

My android app is using AWS Amplify for authentication, data storage & syncing (Cognito, S3, AppSync, API Gateway).

If **user1** is currently logged in to  **Device A**, then **user1** logs in to **Device B**

Questions: 

1. Is there an API/listener to notify **Device A** that the user signs in to a new device?
2. Is it possible to revoke AppSync session/token?

Amplify version: 1.36.1

**Comment by mattcreaser** [comment]
Hi @xnicoting Amplify itself doesn't offer any mechanism to detect multiple sign ins. It may be possible to build something along those lines with Lambda triggers, but it's out of the scope of this library. You could also perhaps do something like storing a unique device ID in AppSync and comparing that to the local ID.

Revoking AppSync access depends on the [authorization mode](https://docs.aws.amazon.com/appsync/latest/devguide/security-authz.html) you're using for AppSync. I don't believe you can revoke access when using `API_KEY`, for example, whereas when using `AWS_LAMBDA` you have complete control. If you're using `AMAZON_COGNITO_USER_POOLS` then signing the user out will prevent accessing AppSync.

Hope that helps!

**Comment by xnicoting** [comment]
Hi @mattcreaser,

Thanks for the response, Actually that's what I did, since I have a `Configuration` table for each account , I saved the device id on it and when user signs in it will compare the `current device id` to the `device id from the cloud`.

The problem is efficiency, so if `user1 `is currently logged in to `Device A` and `user1 `tries to log in to `Device B` there is no way to inform `Device A` that `user1 `logs in to another device. What I did is when `user1 `tries to click specific button from the app(make a transaction button for example) on `Device A`, it will get the latest `device id` from cloud then compares it to current, and if it doesn't match, then `user1` will be sign out from `Device A`. In this solution the problem is the app will call `API.query `every time that button is clicked and this is not efficient since its calling an API and what if we have a million users. Anyway thanks for confirming that there is no API/listener for that. I'll think of a solution using Lamda Functions.

Thanks
